gninasrc/lib/parallel.h

```diff
@@ -59,6 +58,7 @@ struct parallel_for : private boost::thread_group {
     }
 private:
 	void loop(sz offset) {
+    thread_init()();
```

Comment:
What not make thread_init a function (or a function object) so you don't need the double parens?